api/repo.go

```diff
@@ -112,9 +112,7 @@ func CreateRepo(c *gin.Context) {
 		r.SetTimeout(constants.BuildTimeoutMin)
 	}
 
-	if len(input.GetVisibility()) == 0 {
-		r.SetVisibility(constants.VisibilityPublic)
-	}
+	r.SetVisibility(constants.VisibilityPublic)
```

Comment:
I think we're close with this change but I think I'd like to see it modified slightly 👍 

The behavior you're experiencing is due to how the code is written.

Previously, we told the API set the `repo.Visibility` field to `public` **if one was not** provided from the `input`.

The problem is we don't do anything with the `repo.Visibility` field **if one was** provided from the `input`.

Instead, I believe the code should look like:

```suggestion
	if len(input.GetVisibility()) == 0 {
		r.SetVisibility(constants.VisibilityPublic)
	} else {
		r.SetVisibility(input.GetVisibility())
	}
```

Reply:
🤦 looks like i goofed that in a recent change. my bad. good fine @jbrockopp 👍

Reply:
yeah youre right, good call, I mentioned that in the PR description and saw it was in updaterepo